frontend/src/features/manageProfile/pages/UserManagement.tsx

```diff
@@ -49,13 +74,62 @@ export const UserManagement = () => {
     setIsDeleteDialogOpen(() => true);
   }, []);
 
+  /**
+   * Retrieves user identifiers based on their status from the row selection state.
+   * @param {BCeIDUserStatusType} userStatus The status of the users to filter by.
+   * @returns {string[]} An array of user identifiers (either user names or GUIDs)
+   *                     based on the requested status.
+   */
+  const getSelectedUsers = useCallback(
+    (userStatus: BCeIDUserStatusType) =>
+      Object.keys(rowSelection)
+        .filter((value: string) => value.split(",")[1] === userStatus)
+        .map((value: string) => value.split(",")[0]),
+    [rowSelection],
+  );
+
+  /**
+   * Resets the selection list of users and refetches the data.
+   */
+  const resetUsersList = () => {
+    setRowSelection(() => {
+      return {};
+    });
+    query.refetch();
+  };
+
   /**
    * Function that deletes one or more users.
    */
   const onConfirmDelete = async () => {
-    // Uncomment this line -const userNames: string[] = Object.keys(rowSelection);
-    // For implementation.
     setIsDeleteDialogOpen(() => false);
+    const userNames: string[] = getSelectedUsers(BCeID_USER_STATUS.PENDING);
+    const userGUIDs: string[] = getSelectedUsers(BCeID_USER_STATUS.ACTIVE);
+    // Because there are two APIs, one for pending users and one for existing users
+    // We await the response of first before proceeding to the next one.
+    // Promise.all isn't feasible because, API does not accept an empty array.
+    if (userNames.length > 0) {
+      const { data: pendingUserResponse } =
+        await deleteCompanyPendingUsers(userNames);
+      const { failure } = pendingUserResponse as DeleteResponse;
+      if (failure?.length > 0) {
+        navigate(ERROR_ROUTES.UNEXPECTED);
+      }
+    }
+    if (userGUIDs.length > 0) {
+      deleteCompanyUsers(userGUIDs)
```

Comment:
Minor question, but should we use await here instead of then for promise handling? Also should we use a hook with a mutation to encapsulate calling deleteCompanyUsers() call (possibly also invalidating the getCompanyUsers query inside the hook)?